shared/src/stripe/StripeEventHandler.ts

```diff
@@ -112,6 +113,28 @@ export class StripeEventHandler {
 		};
 	};
 
+	/**
+	 * Increments the total donations of a campaign if the charge is associated with a campaignId.
+	 */
+	maybeUpdateCampaign = async (charge: Stripe.Charge): Promise<void> => {
+		if (charge.metadata.campaignId) {
+			const campaignRef = this.firestoreAdmin
+				.collection<Campaign>(CAMPAIGN_FIRESTORE_PATH)
+				.doc(charge.metadata.campaignId);
+
+			try {
+				const campaign = await campaignRef.get();
+				const current_amount_chf = campaign.data()?.amount_collected_chf ?? 0;
+				await campaignRef.update({
+					amount_collected_chf: current_amount_chf + charge.amount,
```

Comment:
Is it possible that you should use `charge.balanceTransaction.amount` here? The `amount` field could be in a different currency, right?

At least that's how we do it in the `constructContribution()` function.

Reply:
Ah very good point. Will change.